solara/toestand.py

```diff
@@ -268,23 +275,93 @@ def set(self, value: S):
     def initial_value(self) -> S:
         pass
 
+    def _check_mutation(self):
+        pass
+
+
+def _find_outside_solara_frame() -> Optional[FrameType]:
+    # the module where the call stack origined from
+    current_frame = inspect.currentframe()
```

Comment:
In interest of performance, should we use `sys._getframe`? Recall the discussion in https://github.com/widgetti/solara/pull/517, which references the testing done in https://gist.github.com/JettJones/c236494013f22723c1822126df944b12.

Reply:
Sound good!